p2p/src/utils/requester/requester.rs

```diff
@@ -166,7 +212,12 @@ impl<E: Clock + GClock + Rng + Metrics, P: PublicKey> Requester<E, P> {
 
         // Increment failed metric if no participants are available
         self.metrics.created.inc(Status::Failure);
-        Err(next)
+
+        if next == Duration::MAX {
+            Err(Error::NoEligibleParticipants)
```

Comment:
We should change `next` to `Option<>`...it is possible `limit.wait_time_from` could return Duration::MAX as well.